Uplift/Views/GymDetail/GymDetailFacilitiesCell.swift

```diff
@@ -7,103 +7,89 @@
-        facilitiesLabel.textColor = .primaryBlack
-        facilitiesLabel.text = ClientStrings.GymDetail.facilitiesSection
-        contentView.addSubview(facilitiesLabel)
-
-        gymFacilitiesTableView = UITableView(frame: .zero, style: .plain)
-        gymFacilitiesTableView.bounces = false
-        gymFacilitiesTableView.showsVerticalScrollIndicator = false
-        gymFacilitiesTableView.separatorStyle = .none
-        gymFacilitiesTableView.backgroundColor = .clear
-        gymFacilitiesTableView.isScrollEnabled = false
-        gymFacilitiesTableView.allowsSelection = false
-        gymFacilitiesTableView.register(GymFacilitiesCell.self, forCellReuseIdentifier: GymFacilitiesCell.identifier)
-        gymFacilitiesTableView.delegate = self
-        gymFacilitiesTableView.dataSource = self
-        contentView.addSubview(gymFacilitiesTableView)
-
-        dividerView.backgroundColor = .gray01
-        contentView.addSubview(dividerView)
+    func configure(for facilities: [Facility], dropdownStatuses: [DropdownStatus], reloadGymDetailCollectionViewClosure: @escaping ((Int) -> ())) {
+        setNeedsUpdateConstraints()
+        self.facilities = facilities
+        self.dropdownCellStatuses = dropdownStatuses
+        self.reloadGymDetailCollectionViewClosure = reloadGymDetailCollectionViewClosure
+        collectionView.reloadData()
     }
 
-    func setupConstraints() {
-        facilitiesLabel.snp.makeConstraints { make in
-            make.top.equalToSuperview().inset(Constraints.verticalPadding)
-            make.centerX.equalToSuperview()
-            make.height.equalTo(Constraints.titleLabelHeight)
+    override func updateConstraints() {
+        super.updateConstraints()
+        collectionView.snp.makeConstraints { make in
+            make.top.leading.trailing.bottom.equalToSuperview()
         }
+    }
 
-        if let gymFacilitiesTableView = gymFacilitiesTableView {
-            gymFacilitiesTableView.snp.remakeConstraints { make in
-                make.top.equalTo(facilitiesLabel.snp.bottom).offset(Constants.gymFacilitiesTopPadding)
-                make.width.centerX.equalToSuperview()
-                make.height.equalTo(Constants.gymFacilitiesCellHeight * CGFloat(gymFacilities.count))
-            }
-
-            dividerView.snp.remakeConstraints { make in
-                make.top.equalTo(gymFacilitiesTableView.snp.bottom).offset(Constraints.verticalPadding)
-                make.leading.trailing.equalToSuperview()
-                make.height.equalTo(Constraints.dividerViewHeight)
+    static func getHeights(for facilities: [Facility], dropdownCellStatuses: [DropdownStatus]?) -> CGFloat {
+        var height: CGFloat = 0
+        for i in 0..<facilities.count {
+            if dropdownCellStatuses?[i] == .open {
+                height += FacilitiesDropdownCell.getHeights(for: facilities[i])
+            } else {
+                height += FacilitiesDropdownCell.headerViewHeight
```

Comment:
you don't really seem to need the index here so can just switch to
```
facilities.forEach { ... }
or
for facility in facilities { ... }
```

Reply:
We use the index to access the corresponding dropdown status in `dropdownCellStatuses` on line 59.